test/widgets/profile_test.dart

```diff
@@ -365,4 +377,270 @@ void main() {
+
+    testWidgets('smoke, turn off', (tester) async {
+      final toggleInteractionMode = toggleInteractionModeVariant.currentValue!;
+
+      await setupPage(tester, pageUserId: eg.selfUser.userId);
+      await store.handleEvent(UserSettingsUpdateEvent(id: 1,
+        property: UserSettingName.presenceEnabled, value: false));
+      await tester.pump();
+      checkAppearsActive(tester, true);
+
+      // The appearance changes and the request is sent, immediately.
+      prepareRequestSuccess(Duration(milliseconds: 100));
+      scheduleEventAfter(Duration(milliseconds: 150), false);
+      await doToggle(tester, toggleInteractionMode);
+      await tester.pump();
+      await tester.pump();
+      checkAppearsActive(tester, false);
+      checkRequest(false);
+
+      // Wait a while, idly: no change, no extra requests
+      await tester.pump(RemoteSettingBuilder.localEchoIdleTimeout * 2);
+      check(connection.takeRequests()).isEmpty();
+      checkAppearsActive(tester, false);
+    }, variant: toggleInteractionModeVariant);
+
+    testWidgets('event arrives after local-echo timeout', (tester) async {
+      final toggleInteractionMode = toggleInteractionModeVariant.currentValue!;
+
+      await setupPage(tester, pageUserId: eg.selfUser.userId);
+      check(store.userSettings.presenceEnabled).isTrue();
+      checkAppearsActive(tester, false);
+
+      // The appearance changes and the request is sent, immediately.
+      prepareRequestSuccess(Duration(milliseconds: 100));
+      scheduleEventAfter(Duration(seconds: 10), true);
+      await doToggle(tester, toggleInteractionMode);
+      await tester.pump();
+      await tester.pump();
+      checkAppearsActive(tester, true);
+      checkRequest(true);
+
+      // Local-echo timeout passes and event hasn't come; change back.
+      await tester.pump(RemoteSettingBuilder.localEchoIdleTimeout);
+      await tester.pump();
+      checkAppearsActive(tester, false);
+
+      // The event comes after a while; update for the new value.
+      await tester.pump(Duration(seconds: 10));
+      check(connection.takeRequests()).isEmpty();
+      checkAppearsActive(tester, true);
+    }, variant: toggleInteractionModeVariant);
+
+    testWidgets('request has an error', (tester) async {
+      final toggleInteractionMode = toggleInteractionModeVariant.currentValue!;
+
+      await setupPage(tester, pageUserId: eg.selfUser.userId);
+      check(store.userSettings.presenceEnabled).isTrue();
+      checkAppearsActive(tester, false);
+
+      // The appearance changes and the request is sent, immediately.
+      prepareRequestError(Duration(milliseconds: 100));
+      await doToggle(tester, toggleInteractionMode);
+      await tester.pump();
+      await tester.pump();
+      checkAppearsActive(tester, true);
+      checkRequest(true);
+
+      // After [PerAccountSettingBuilder.localEchoMinimum], changes back
+      await tester.pump(RemoteSettingBuilder.localEchoMinimum);
```

Comment:
Similarly, this should gain a check that the appearance doesn't change *before* that minimum (when the request failure comes back).